Vendor code from Go standard library.

russjones opened the pull request:
**Description**

Don't use `strings.ReplaceAll` (introduced in Go 1.12) or `strings.Builder` (introduced in Go 1.10) which break Go 1.9.7 builds.

Note that strings.Builder can not be vendored because "go vet" fails on it (causing our build to fail) until the following is resolved: golang/go#23382

russjones committed:
Use Go 1.9.7 friendly functions.

Don't use strings.ReplaceAll (introduced in Go 1.12) or strings.Builder
(introduced in Go 1.10) which break Go 1.9.7 builds.

Note that strings.Builder can not be vendored because "go vet" fails on
it (causing our build to fail) until the following is resolved:
https://github.com/golang/go/issues/23382

fspmarshall reviewed (approved):
Use of `unsafe.Pointer` is angering `go vet` (and therefore jenkins).  LGTM otherwise.

russjones commented:
@fspmarshall Yes, looks like we'll have to fallback to `bytes.Buffer` which I've done and updated the PR. Updated the description with details as well.

russjones commented:
@gravitational-jenkins retest this please

russjones commented:
@gravitational-jenkins retest this please

russjones commented:
@gravitational-jenkins retest this please
